### PR TITLE
lint task: update scripts/eslint.js sharing code with linc.js;

### PR DESCRIPTION
**Pull request by sjy**
Use eslint's CLIEngine to run lint task;

### CODE REVIEW

**Comment by xjlim** [comment]
@sjy I believe you need to combined both scripts.

File: scripts/tasks/eslint.js [hunk 1]
```diff
@@ -7,24 +7,17 @@
...
-    console.error('Lint failed');
-  } else {
-    console.log('Lint passed');
-  }
+const report = cli.executeOnFiles(['.']);
+console.log(formatter(report.results));
 
-  process.exit(code);
-});
+if (report.errorCount > 0) {
```
Comment (xjlim) [inline scripts/tasks/eslint.js]: Should also check for zero warnings.

**Comment by xjlim** [comment]
Build failing, run `npm run prettier`.

**Comment by sjy** [comment]
@xjlim  thanks for your comment, the previous ci failed because I forgot to ran `yarn prettier-all`.

**Comment by xjlim** [comment]
I had initially thought of this as a script that will do either task given a set argument cc @gaearon

**Comment by gaearon** [comment]
This is great. Thanks!

**Pull request merged**